rush-plugins/rush-azure-storage-build-cache-plugin/src/ChainedCredential.ts

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+import {
+  ChainedTokenCredential,
+  VisualStudioCodeCredential,
+  AzureCliCredential,
+  AzureDeveloperCliCredential,
+  AzurePowerShellCredential,
+  InteractiveBrowserCredential,
+  DeviceCodeCredential
+} from '@azure/identity';
+import type { TokenCredentialOptions } from '@azure/identity';
+import { AdoCodespacesAuthCredential } from './AdoCodespacesAuthCredential';
+
+export class ChainedCredential extends ChainedTokenCredential {
+  public constructor(options: TokenCredentialOptions) {
+    super(
+      new AdoCodespacesAuthCredential(),
+      new VisualStudioCodeCredential(options),
+      new AzureCliCredential(options),
+      new AzureDeveloperCliCredential(options),
+      new AzurePowerShellCredential(options),
+      new InteractiveBrowserCredential(options),
+      new DeviceCodeCredential(options)
```

Comment:
It'd be useful to 1) expose all of these to developers and repo maintainers and 2) allow the order and inclusion/exclusion to be customized.

Reply:
To make it more flexible, we could allow the developers to provide their own implementation of `TokenCredential`. This way they can implement their own chaining logic or implement custom credentials like we have the `AdoCodespacesAuthCredential` custom `TokenCredential` for rushstack.
Thoughts?
Or we can add an array that has a default value just for the chained credential.

Reply:
We could just implement `loginFlowFailover` by invoking `new ChainedTokenCredential(...credentials)`